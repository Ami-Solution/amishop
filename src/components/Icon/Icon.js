import React from 'react';
import styled, { css } from 'styled-components';
import PropTypes from 'prop-types';

const Svg = styled.svg`
  display: block;
  overflow: visible;
  ${({ size }) => css`
    width: ${size};
    height: ${size};
  `};
`;

const Line = styled.line.attrs({
  vectorEffect: 'non-scaling-stroke',
})`
  stroke: ${({ theme }) => theme.color};
  stroke-width: ${({ theme }) => theme.strokeWidth};
  fill: ${({ theme }) => theme.fill};
`;

const Polyline = Line.withComponent('polyline');

const icons = {
  close: theme => (
    <React.Fragment>
      <Line theme={theme} x1="0" y1="0" x2="12" y2="12" />
      <Line theme={theme} x1="0" y1="12" x2="12" y2="0" />
    </React.Fragment>
  ),
  prev: theme => (
    <React.Fragment>
      <Line theme={theme} x1="0" y1="6" x2="12" y2="6" />
      <Polyline theme={theme} points="6 0 0 6 6 12" />
    </React.Fragment>
  ),
  carretDown: theme => <Polyline theme={theme} points="0 3 6 9 12 3" />,
  carretUp: theme => <Polyline theme={theme} points="0 9 6 3 12 9" />,
  carretRight: theme => <Polyline theme={theme} points="3 0 9 6 3 12" />,
  check: theme => <Polyline theme={theme} points="0 6 4 10 12 1" />,
};

const Icon = ({ size, name, color, strokeWidth, fill }) => (
  <Svg viewBox="0 0 12 12" size={size}>
    {icons[name]({ color, strokeWidth, fill })}
  </Svg>
);

Icon.propTypes = {
  size: PropTypes.string,
  color: PropTypes.string,
  strokeWidth: PropTypes.string,
  fill: PropTypes.string,
  name: PropTypes.oneOf(Object.keys(icons)).isRequired,
};

Icon.defaultProps = {
  size: '1.2rem',
  color: '#000',
  strokeWidth: '0.1rem',
  fill: 'none',
};

export default Icon;
