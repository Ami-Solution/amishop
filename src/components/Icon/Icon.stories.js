/* eslint-disable */
import React from 'react'
import styled, { css } from 'styled-components'
import { storiesOf } from '@storybook/react'
import Icon from './Icon'

storiesOf('03 - Components', module)
.addWithChapters('Icons', {
  chapters: [
    {
      sections: [
        {
          title: 'check icon',
          sectionFn: () => <Icon name="check" strokeWidth="0.2rem"/>,
          options: {
            showSource: true,
            showPropTables: false
          }
        },{
          title: 'close icon',
          sectionFn: () => <Icon name="close" />,
          options: {
            showSource: true,
            showPropTables: false
          }
        },{
          title: 'prev icon',
          sectionFn: () => <Icon name="prev" />,
          options: {
            showSource: true,
            showPropTables: false
          }
        },{
          title: 'carretDown icon',
          sectionFn: () => <Icon name="carretDown" />,
          options: {
            showSource: true,
            showPropTables: false
          }
        },{
          title: 'carretUp icon',
          sectionFn: () => <Icon name="carretUp"/>,
          options: {
            showSource: true,
            showPropTables: false
          }
        },{
          title: 'carretRight icon',
          sectionFn: () => <Icon name="carretRight"/>,
          options: {
            showSource: true,
            showPropTables: false
          }
        }
      ]
    }
  ]
})
