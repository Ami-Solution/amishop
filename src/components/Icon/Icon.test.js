/* global describe it expect */
import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import 'jest-styled-components';

import Icon from './index';

configure({ adapter: new Adapter() });

describe('Component Icon', () => {
  it('should render check', () => {
    const component = renderer.create(<Icon name="check" />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render prev', () => {
    const component = renderer.create(<Icon name="prev" />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render carretDown', () => {
    const component = renderer.create(<Icon name="carretDown" />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render carretUp', () => {
    const component = renderer.create(<Icon name="carretUp" />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render close', () => {
    const component = renderer.create(<Icon name="close" />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
