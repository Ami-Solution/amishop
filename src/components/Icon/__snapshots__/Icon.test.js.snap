// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Icon should render carretDown 1`] = `
.c0 {
  display: block;
  overflow: visible;
  width: 1.2rem;
  height: 1.2rem;
}

.c1 {
  stroke: #000;
  stroke-width: 0.1rem;
  fill: none;
}

<svg
  className="c0"
  size="1.2rem"
  viewBox="0 0 12 12"
>
  <polyline
    className="c1"
    points="0 3 6 9 12 3"
    vectorEffect="non-scaling-stroke"
  />
</svg>
`;

exports[`Component Icon should render carretUp 1`] = `
.c0 {
  display: block;
  overflow: visible;
  width: 1.2rem;
  height: 1.2rem;
}

.c1 {
  stroke: #000;
  stroke-width: 0.1rem;
  fill: none;
}

<svg
  className="c0"
  size="1.2rem"
  viewBox="0 0 12 12"
>
  <polyline
    className="c1"
    points="0 9 6 3 12 9"
    vectorEffect="non-scaling-stroke"
  />
</svg>
`;

exports[`Component Icon should render check 1`] = `
.c0 {
  display: block;
  overflow: visible;
  width: 1.2rem;
  height: 1.2rem;
}

.c1 {
  stroke: #000;
  stroke-width: 0.1rem;
  fill: none;
}

<svg
  className="c0"
  size="1.2rem"
  viewBox="0 0 12 12"
>
  <polyline
    className="c1"
    points="0 6 4 10 12 1"
    vectorEffect="non-scaling-stroke"
  />
</svg>
`;

exports[`Component Icon should render close 1`] = `
.c0 {
  display: block;
  overflow: visible;
  width: 1.2rem;
  height: 1.2rem;
}

.c1 {
  stroke: #000;
  stroke-width: 0.1rem;
  fill: none;
}

<svg
  className="c0"
  size="1.2rem"
  viewBox="0 0 12 12"
>
  <line
    className="c1"
    vectorEffect="non-scaling-stroke"
    x1="0"
    x2="12"
    y1="0"
    y2="12"
  />
  <line
    className="c1"
    vectorEffect="non-scaling-stroke"
    x1="0"
    x2="12"
    y1="12"
    y2="0"
  />
</svg>
`;

exports[`Component Icon should render prev 1`] = `
.c0 {
  display: block;
  overflow: visible;
  width: 1.2rem;
  height: 1.2rem;
}

.c1 {
  stroke: #000;
  stroke-width: 0.1rem;
  fill: none;
}

.c2 {
  stroke: #000;
  stroke-width: 0.1rem;
  fill: none;
}

<svg
  className="c0"
  size="1.2rem"
  viewBox="0 0 12 12"
>
  <line
    className="c1"
    vectorEffect="non-scaling-stroke"
    x1="0"
    x2="12"
    y1="6"
    y2="6"
  />
  <polyline
    className="c2"
    points="6 0 0 6 6 12"
    vectorEffect="non-scaling-stroke"
  />
</svg>
`;
